substrate/frame/scheduler/src/lib.rs

```diff
@@ -440,6 +462,86 @@ pub mod pallet {
 			)?;
 			Ok(())
 		}
+
+		/// Set a retry configuration for a task so that, in case its scheduled run fails, it will
+		/// be retried after `period` blocks, for a total amount of `retries` retries or until it
+		/// succeeds.
+		///
+		/// If a retried task runs successfully before running out of retries, its remaining retry
+		/// counter will be reset to the initial value. If a retried task runs out of retries, it
+		/// will be removed from the schedule.
+		///
+		/// Tasks which need to be scheduled for a retry are still subject to weight metering and
+		/// agenda space, same as a regular task. Periodic tasks will have their periodic schedule
+		/// put on hold while the task is retrying.
+		#[pallet::call_index(6)]
+		#[pallet::weight(<T as Config>::WeightInfo::set_retry(T::MaxScheduledPerBlock::get()))]
+		pub fn set_retry(
+			origin: OriginFor<T>,
+			when: BlockNumberFor<T>,
+			index: u32,
```

Comment:
I think there is a convenience type to reference a task: `TaskAddress<BlockNumberFor<T>>`

Reply:
I saw that but I decided to go with the current convention in the code, as seen in [`cancel`](https://github.com/paritytech/polkadot-sdk/blob/73fd8cd717c4df533e3c4fbaa2ee4f06504bf978/substrate/frame/scheduler/src/lib.rs#L357).

Reply:
But then in the event its a `TaskAddress`? I think it should be consistent...

Reply:
Changed it to use `TaskAddress`.